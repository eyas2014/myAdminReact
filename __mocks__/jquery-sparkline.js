$.fn.sparkline=function (args){
	$.calledWithResult.push(args);
}