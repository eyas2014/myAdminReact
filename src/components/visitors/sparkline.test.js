import React from 'react';
import 'jquery-sparkline';
import enzyme from 'enzyme';
import SparkLine from './sparkline.jsx';

jest.mock('jquery-sparkline');

document.body.innerHTML='<div  id="root"></div>';

var data=[1,2,3];

it('renders correct htmls',()=>{
	const wrapper=enzyme.shallow(<SparkLine data={data}></SparkLine>);
	expect(wrapper.html()).toMatchSnapshot();
});

beforeEach(()=>{$.isCalledWith=[]});

it('$.sparkline should be called with correct arguments', ()=>{
	const wrapper=enzyme.mount(<SparkLine data={data}></SparkLine>, {attachTo: document.getElementById('root')});
	expect($.isCalledWith).toEqual(data);
});