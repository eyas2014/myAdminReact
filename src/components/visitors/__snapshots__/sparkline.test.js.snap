// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct htmls 1`] = `"<div class=\\"row\\"><div class=\\"col-4 text-center text-muted\\"><span id=\\"sparkline1\\"></span><span>Visitors</span></div><div class=\\"col-4 text-center text-muted\\"><span id=\\"sparkline2\\"></span><span>Online</span></div><div class=\\"col-4 text-center text-muted\\"><span id=\\"sparkline3\\"></span><span>Exists</span></div></div>"`;
